elastic/kibana
# [Discover] Fix loading of a single doc JSON

Closes https://github.com/elastic/kibana/issues/131785

## Summary

This PR changes how we request data for a single doc. Now it passes index name as a query for ES.

How to test:
- Create an index via DevTools with a `.` in front: ex. `.recipes` (as of now indices with a dot are considered to be hidden and they would not be accessible to viewer-users) or control access to this index via custom privileges 
- Create an alias for this new index `PUT .recipes/_alias/recipes_alias_test`
- Create a data view with `recipes_alias_test` as a pattern 
- Create a new user with "viewer" role and login with that user
- Go to Discover and choose the data view
- Check that grid is rendered and that a single document can be opened too (in flyout and as a page) and its JSON is available.

Before:
![Jun-29-2022 17-54-23](https://user-images.githubusercontent.com/1415710/176481484-6f29b4be-248c-4470-9fb6-ee2ea219baea.gif)

After:
![Jun-29-2022 17-52-06](https://user-images.githubusercontent.com/1415710/176481558-554ade13-6786-4b3d-996f-35972c164df1.gif)

This PR also transforms the Single Doc and Surrounding Docs buttons into links so they can be opened in a new tab.
<img width="807" alt="Screenshot 2022-06-30 at 13 03 06" src="https://user-images.githubusercontent.com/1415710/176662116-dda2c158-457d-4d8d-b6f7-bfec5038085d.png">
 


### Checklist
